dallinger/experiment.py

```diff
@@ -664,6 +667,166 @@ def is_complete(self):
         """
         return None
 
+    def monitoring_panels(self, **kw):
+        """Provides monitoring dashboard sidebar panels.
+
+        :param \**kw: arguments passed in from the request
+        :returns: An ``OrderedDict()`` mapping panel titles to HTML strings
+                  to render in the dashboard sidebar.
+        """  # noqa
+        stats = self.monitoring_statistics(**kw)
+        panels = OrderedDict()
+        for tab in stats:
+            panels[tab] = struct_to_html(stats[tab])
```

Comment:
Is it that tightly bound? IIRC, the only thing specific to this data is the special case of a two element dict containing the keys "count" and "failed", and even that is rendered in a fairly generic way.

I separated the two methods because there may be cases where people want to add data to the statistics or otherwise monitor the data, but use continue to use the default HTML rendering, but the much more common case is where people want to add their own panels and or append custom HTML to existing panels. So I don't really expect people to customize `monitoring_statistics` as much as I would `monitoring_panels`, but I put it in the public API just in case people might find value in doing so.

Reply:
Also, the main reason I separated out `struct_to_html ` and `monitoring_statistics` was to keep the code in `monitoring_panels` very simple and easy to understand, since that's the method I think people are most likely to want to override to provide/extend custom HTML panels.

Reply:
I'm mostly convinced. It's a framework for monitoring things with counts and failed flags, which will almost always be what someone wants to monitor.